src/System.Runtime/tests/System/NullableMetadataTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Runtime.Tests
+{
+    public class NullableMetadataTests
+    {
+        private const string NullableAttributeFullName = "System.Diagnostics.CompilerServices.NullableAttribute";
+        private const string NullableContextAttributeFullName = "System.Diagnostics.CompilerServices.NullableContextAttribute";
+        private const string NullablePublicOnlyAttributeFullName = "System.Diagnostics.CompilerServices.NullablePublicOnlyAttribute";
+
+        [Fact]
+        public static void NullableAttributesOnPublicApiOnly()
+        {
+            MemberInfo[] internalMembers = typeof(string).GetMembers(BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance);
+
+            foreach (MemberInfo internalMember in internalMembers)
+            {
+                Assert.Empty(internalMember.GetCustomAttributes().Where(a => NullableAttributesFilter(a)));
+            }
+        }
```

Comment:
Can we also validate that a nullable attribute shows up in at least one place?

Reply:
Sure. I'll validate we at least have a `NullableAttribute` and a `NullableContextAttribute`.